main.go

```diff
@@ -272,3 +287,37 @@ func (m *metricHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		closer.Close()
 	}
 }
+
+func detectSharding(kubeClient clientset.Interface, podName, namespaceName string) (shard, totalShards uint64, err error) {
+	glog.Infof("Auto detecting shard.")
+	p, err := kubeClient.CoreV1().Pods(namespaceName).Get(podName, metav1.GetOptions{})
+	if err != nil {
+		return 0, 0, err
```

Comment:
nit: let's add some context to indicate where we are, i.e. `fmt.Errorf("error detecting shard: %v", err)`